techmatters/flex-plugins
# CHI-1234: Copy Household

Primary reviewer: @murilovmachado 

## Description

* Adds a new 'copy-to' type of form element, with a 'savable' flag set to false to indicate it shouldn't be persisted to the form.
* Refactors section specific transformation / update / find operations into a separate API interface
* Added logic to copy a case item to another section when a 'copy-to' checkbox is selected

### Checklist
- N/A Strings are localized
- N/A Tested for call contacts

### Related Issues
Fixes CHI-1234

### Verification steps

In Aselo Development

1. Create a household
2. Populate all the fields
3. Tick the 'Also a perpetrator' box
4. Check a perpetrator is created with all the same details

Repeat the above steps WITHOUT checking the box, ensure nothing is created
Repeat the above steps but create a perpetrator and copy it to a household

We can also test edge cases not present in the baseline demo forms by either running flex locally and editing the demo definition, or by deploying it to staging environments where different perp / household member defintions exist
